metadata.rb

```diff
@@ -10,9 +10,9 @@
 
 depends          'base'
 depends          'build-essential'
-depends          'git', '~> 4.3'
+depends          'git'
 depends          'java'
-depends          'jenkins', '~> 2.4.0'
+depends          'jenkins', '~> 5.0.0'
 depends          'osl-haproxy'
 depends          'ssh-keys'
```

Comment:
@knightsamar I agree we should be pinning at least minor versions which is what this is doing. I've been mainly adding it when I can and where it makes sense.

@jldugger the uploader won't upload any deps automatically. With the new checks I added to the chef-repo jenkins job, that should catch any dep issues we may run into.